packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/eventbridge-scheduler/create-schedule.ts

```diff
@@ -0,0 +1,364 @@
+import { Construct } from 'constructs';
+import * as iam from '../../../aws-iam';
+import * as kms from '../../../aws-kms';
+import * as sqs from '../../../aws-sqs';
+import * as sfn from '../../../aws-stepfunctions';
+import { Stack, Duration, Token } from '../../../core';
+import { integrationResourceArn } from '../private/task-utils';
+import { Schedule } from '../schedule';
+
+/**
+ * The action that EventBridge Scheduler applies to the schedule after the schedule completes invoking the target.
+ */
+export enum ActionAfterCompletion {
+  /**
+   * Takes no action
+   */
+  NONE = 'NONE',
+  /**
+   * Deletes the schedule
+   */
+  DELETE = 'DELETE',
+}
+
+/**
+ * Properties for creating an AWS EventBridge Scheduler schedule
+ */
+export interface EventBridgeSchedulerCreateScheduleTaskProps extends sfn.TaskStateBaseProps {
+  /**
+   * Schedule name
+   */
+  readonly scheduleName: string;
+
+  /**
+   * Specifies the action that EventBridge Scheduler applies to the schedule after the schedule completes invoking the target.
+   *
+   * @default ActionAfterCompletion.NONE
+   */
+  readonly actionAfterCompletion?: ActionAfterCompletion;
+
+  /**
+   * Unique, case-sensitive identifier to ensure the idempotency of the request.
+   *
+   * @default - Automatically generated
+   */
+  readonly clientToken?: string;
+
+  /**
+   * The description for the schedule.
+   *
+   * @default - No description
+   */
+  readonly description?: string;
+
+  /**
+   * The date, in UTC, before which the schedule can invoke its target.
+   * Depending on the schedule's recurrence expression, invocations might stop on, or before, the EndDate you specify.
+   * EventBridge Scheduler ignores EndDate for one-time schedules.
+   *
+   * @default - No end date
+   */
+  readonly endDate?: Date;
+
+  /**
+   * The maximum time window during which a schedule can be invoked.
+   *
+   * Minimum value is 1 minute.
+   * Maximum value is 1440 minutes (1 day).
+   *
+   * @default - Flexible time window is not enabled.
+   */
+  readonly flexibleTimeWindow?: Duration;
+
+  /**
+   * The name of the schedule group to associate with this schedule.
+   *
+   * @default - The default schedule group is used.
+   */
+  readonly groupName?: string;
+
+  /**
+   * The customer managed KMS key that EventBridge Scheduler will use to encrypt and decrypt payload.
+   *
+   * @see https://docs.aws.amazon.com/scheduler/latest/UserGuide/encryption-rest.html
+   *
+   * @default - use automatically generated KMS key
```

Comment:
```suggestion
   * @default - Use automatically generated KMS key
```
All the other `@default`s are capped 🫠